bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+
+import numpy as np
+import h5py
+import operator
+import argparse
+import logging
+from awkde import GaussianKDE
+from pycbc import init_logging
+import pycbc.conversions as convert
+from pycbc import libutils
+kf = libutils.import_optional('sklearn.model_selection')
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--signal-file')
+parser.add_argument('--template-file', required=True)
+parser.add_argument('--output-file', required=True)
+parser.add_argument('--make-signal-kde', action='store_true')
+parser.add_argument('--make-template-kde', action='store_true')
+parser.add_argument('--verbose', action='count')
+args = parser.parse_args()
+init_logging(args.verbose)
+
+def kde_awkde(x, x_grid, alp=0.5, gl_bandwidth='silverman', ret_kde=False):
+    kde = GaussianKDE(glob_bw=gl_bandwidth, alpha=alp, diag_cov=True)
+    kde.fit(x)
+    if isinstance(x_grid, (list, tuple, np.ndarray)) == False:
+        y = kde.predict(x_grid)
+    else:
+        y = kde.predict(x_grid)
+    if ret_kde == True:
+        return kde, y
+    return y
+
+def kfcv_awkde(sample, bwchoice, alphachoice, k=2):
+    if bwchoice not in ['silverman', 'scott']:
+        bwchoice = float(bwchoice)
+    fomlist = []
+    kfold = kf.KFold(n_splits=k, shuffle=True, random_state=None )
+    for train_index, test_index in kfold.split(sample):
+        train, test = sample[train_index], sample[test_index]
+        y = kde_awkde(train, test, alp=alphachoice, gl_bandwidth=bwchoice)
+        fomlist.append(np.sum(np.log(y)))
+    return np.mean(fomlist)
+
+def optimizedparam(sampleval, kval=2):
```

Comment:
I would suggest the variable name 'nfold' rather than 'kval' (k is the 'number of folds' in the K-fold cross validation), otherwise this seems ok